test/uri_util_test.rb

```diff
@@ -0,0 +1,75 @@
+require File.expand_path('../support/test_helper', __FILE__)
+
+class UriUtilTest < Minitest::Test
+  def test_ref_fragment_path
+    uri = '#some-thing'
+    base = 'http://www.example.com/foo/#bar'
+
+    assert_equal Addressable::URI.parse('http://www.example.com/foo/#some-thing'), JSON::Util::URI.normalize_ref(uri, base)
+    assert_equal Addressable::URI.parse('http://www.example.com/foo/#'), JSON::Util::URI.absolutize_ref(uri, base)
+  end
+
+  def test_ref_file_path
+    uri = '/some/thing'
+    base = 'http://www.example.com/foo/#bar'
+
+    assert_equal Addressable::URI.parse('http://www.example.com/some/thing#'), JSON::Util::URI.normalize_ref(uri, base)
+    assert_equal Addressable::URI.parse('http://www.example.com/some/thing#'), JSON::Util::URI.absolutize_ref(uri, base)
+  end
+
+  def test_ref_uri
+    uri = 'http://foo-bar.com'
+    base = 'http://www.example.com/foo/#bar'
+
+    assert_equal Addressable::URI.parse('http://foo-bar.com/#'), JSON::Util::URI.normalize_ref(uri, base)
+    assert_equal Addressable::URI.parse('http://foo-bar.com/#'), JSON::Util::URI.absolutize_ref(uri, base)
+  end
+
+  def test_ref_uri_with_path
+    uri = 'http://foo-bar.com/some/thing'
+    base = 'http://www.example.com/foo/#bar'
+
+    assert_equal Addressable::URI.parse('http://foo-bar.com/some/thing#'), JSON::Util::URI.normalize_ref(uri, base)
+    assert_equal Addressable::URI.parse('http://foo-bar.com/some/thing#'), JSON::Util::URI.absolutize_ref(uri, base)
+  end
+
+  def test_ref_uri_with_fragment
+    uri = 'http://foo-bar.com/some/thing#foo'
+    base = 'http://www.example.com/foo/#bar'
```

Comment:
I think this test case and the one below would have improved readability if the fragment of the first and the path of the second URI would not both be `foo`.